### DISCUSSION TITLE
putListingsItem body schema

### DISCUSSION THREAD

**Discussion by trycatchsoft** [amzn/selling-partner-api-models#3169] [General]
Hello everyone,

I want to use putlistingsitem operation for listing inventory on Amazon. But unfortunately i can not find the requests body schema. I dont know how it seems like. Can please someone who has an example request show me the request body? 

Thanks in advance

**Comment by vaidas-lungis** [comment]
I guess this guide is a good starting point [Listings Items API Use Case Guide](https://github.com/amzn/selling-partner-api-docs/blob/main/guides/en-US/use-case-guides/listings-items-api-use-case-guide/listings-items-api-use-case-guide_2020-09-01.md#what-is-the-listings-items-api)


you need to fetch definition file based on productType (another guide [Product Type Definitions API Use Case Guide](https://github.com/amzn/selling-partner-api-docs/blob/main/guides/en-US/use-case-guides/product-type-definitions-api-use-case-guide/definitions-product-types-api-use-case-guide_2020-09-01.md))
 and prepare a payload

**Reply by trycatchsoft** [reply]
Thank you Vaidas. Is there any method for getting product type with ASIN code?

**Reply by vaidas-lungis** [reply]
I don't think so :/

**Reply by trycatchsoft** [reply]
:(

**Reply by ilguala** [reply]
> Thank you Vaidas. Is there any method for getting product type with ASIN code?

You can search product with ASIN with Catalog API. 

What i don't know is how to put listing item related with an existing ASIN amazon item. Where is the property ASIN than in XML POST_PRODUCT_DATA exists?

**Reply by uk-2** [reply]
can someone help us with this too  @chapmanjw 
How to add a existing offer to ASIN

We are using  putListingsItem with LISTING_OFFER_ONLY
